bambi/models.py

```diff
@@ -27,81 +27,148 @@ class Model:
+        to be treated as categoricals (e.g., group specific factors coded as numerical IDs),
+        explicitly passing variable names via this argument is recommended.
+    dropna : bool
+        When ``True``, rows with any missing values in either the predictors or outcome are
+        automatically dropped from the dataset in a listwise manner.
+    auto_scale : bool
+        If ``True`` (default), priors are automatically rescaled to the data
+        (to be weakly informative) any time default priors are used. Note that any priors
+        explicitly set by the user will always take precedence over default priors.
     default_priors : dict or str
         An optional specification of the default priors to use for all model terms. Either a
         dictionary containing named distributions, families, and terms (see the documentation in
         ``priors.PriorFactory`` for details), or the name of a JSON file containing the same
         information.
-    auto_scale : bool
-        If ``True`` (default), priors are automatically rescaled to the data
-        (to be weakly informative) any time default priors are used. Note that any priors
-        explicitly set by the user will always take precedence over default priors.
-    dropna : bool
-        When ``True``, rows with any missing values in either the predictors or outcome are
-        automatically dropped from the dataset in a listwise manner.
+    noncentered : bool
+        If ``True`` (default), uses a non-centered parameterization for normal hyperpriors on
+        grouped parameters. If ``False``, naive (centered) parameterization is used.
     taylor : int
         Order of Taylor expansion to use in approximate variance when constructing the default
         priors. Should be between 1 and 13. Lower values are less accurate, tending to undershoot
         the correct prior width, but are faster to compute and more stable. Odd-numbered values
         tend to work better. Defaults to 5 for Normal models and 1 for non-Normal models. Values
         higher than the defaults are generally not recommended as they can be unstable.
-    noncentered : bool
-        If ``True`` (default), uses a non-centered parameterization for normal hyperpriors on
-        grouped parameters. If ``False``, naive (centered) parameterization is used.
     """
 
     # pylint: disable=too-many-instance-attributes
     def __init__(
         self,
+        formula=None,
         data=None,
-        default_priors=None,
-        auto_scale=True,
+        family="gaussian",
+        priors=None,
+        link=None,
+        categorical=None,
         dropna=False,
-        taylor=None,
+        auto_scale=True,
+        default_priors=None,
         noncentered=True,
+        taylor=None,
     ):
+        # attributes that are set later
+        self.terms = {}
+        self.dm_statistics = None  # build()
+        self._diagnostics = None  # build()
```

Comment:
I think we can safely remove this

Reply:
`self.terms` must be an empty dictionary because `self._add_common` and `self._add_group_specific` assume it exists, i.e. they do `self.terms[name] = Term(name, term, data, prior)` and `self.terms[name] = GroupSpecificTerm(name, term, data, prior)`

Reply:
But I can delete `self._diagnostics` and `self.dm_statistics`. The former is not used, the latter is created in `build()`

Reply:
Sorry I was talking only about `self._diagnostics`